### PR TITLE
feature(comments): show a general link to the comment section of an entity

### LINKED ISSUE #7356: FR: general comment entity menu item
In the river a comment item is added to the menu if $object->canComment(),

why is this not added to the entity menu?
- [x] add general entity menu hook to add menu item
- [x] add a general id to the comment listing so the link can jump to the comments

### PR DESCRIPTION
**Pull request by jeabakker**
For easy navigation a general entity menu item was added to link to the
comment section of an entity

fixes #7356

### CODE REVIEW

File: engine/lib/comments.php [hunk 1]
```diff
@@ -180,6 +181,44 @@ function _elgg_comment_url_handler($hook, $type, $return, $params) {
...
+ * @access private
+ */
+function _elgg_comments_entity_menu($hook, $type, $return, $params) {
+	
+	if (elgg_in_context('widgets')) {
+		return $return;
+	}
+
+	$entity = elgg_extract('entity', $params);
+	if (!elgg_instanceof($entity)) {
```
Comment (mrclay) [inline engine/lib/comments.php]: For simple ElggEntity checks, `instanceof ElggEntity` is preferred.

File: engine/lib/comments.php [hunk 2]
```diff
@@ -180,6 +181,44 @@ function _elgg_comment_url_handler($hook, $type, $return, $params) {
...
+		return $return;
+	}
+
+	if (!$entity->canComment()) {
```
Comment (mrclay) [inline engine/lib/comments.php]: I'm concerned this will be expensive in listings.

**Comment by mrclay** [comment]
Considering it's easy to remove if unwanted, seems like a good idea. Have not tested it yet. This could go on 1.x.

**Comment by juho-jaakkola** [comment]
I feel that the entity menu is a bit crowded as is. So I'd prefer not to have this by default.

**Comment by mrclay** [comment]
Would it be better as an icon like in the river?

**Comment by juho-jaakkola** [comment]
This PR does add an icon.

But I think that when viewing e.g. a single blog post it is easy to just scroll down a bit (or use End button) to get to the comment section. In the river it is different because there isn't just one item that can be commented but multiple.

**Comment by mrclay** [comment]
Yeah, this needn't appear on the entity's page. I don't know how we'd do that.

**Comment by jeabakker** [comment]
> Yeah, this needn't appear on the entity's page. I don't know how we'd do that.

If we'd pass 'full_view' to `elgg_view_menu` this could be done, but that requires a massive rewrite. I think that's too much for this PR

**Pull request closed without merging**